src/core/test/system/borrower/balance.t.sol

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "../system.sol";
+import "../users/borrower.sol";
+
+contract BalanceTest is SystemTest {
+
+    Borrower borrower;
+    address borrower_;
+        
+    function setUp() public {
+        baseSetup();
+        // setup users
+        borrower = new Borrower(address(shelf), address(distributor), currency_, address(pile));
+        borrower_ = address(borrower);
+    }
+    
+    function balanceTake() public {
+        uint initialCurrencyBalance = currency.balanceOf(address(shelf));
+        uint takeAmount = currency.balanceOf(address(junior));
+        borrower.balance();
+        assertPostConditionTake(takeAmount, initialCurrencyBalance);
+    }
+
+    function balanceGive() public {
+        uint initialCurrencyBalance = currency.balanceOf(address(shelf));
+        uint giveAmount = currency.balanceOf(address(junior));
+        borrower.balance();
+        assertPostConditionGive(giveAmount, initialCurrencyBalance);
+    }
+
+    function assertPreConditionTake(uint takeAmount) public {
+        // assert: borrowFromTranches is active
+        assert(distributor.borrowFromTranches());
+        // assert: junior has enough capital
+        assert(currency.balanceOf(address(junior)) == takeAmount);
```

Comment:
:  >=

Reply:
to test for surplus cases, only enough is transferred to cover debts

Reply:
in case of switchable distributor we always take and give the full available amount

Reply:
I think we should be explicit about that it is testing the switchable distributor setup